### DISCUSSION TITLE
PAG lib是否支持在子线程中加载动画资源？

### DISCUSSION THREAD

**Discussion by aSmartTortoise** [Tencent/libpag#2474] [Q&A] [answered]
众说周知，Android 原生的SurfaceView也可以实现动画，且很好地避免了View在执行动画过程中对主线程的负载，请问PAG lib是否有对应的组件（控件）能实现类似原生的SurfaceView在子线程中加载动画？

**Comment by kevingpqi123** [comment]
PAGView 和 PAGImageView 默认就是在子线程中渲染，只有上屏在主线程，你这边遇到的具体问题是啥

**Reply by aSmartTortoise** [reply]
我这边使用社区版的sdk，使用PAGImageView，发现有概率PAGImageView 不可见。另外PAGImageView是View的子类，如你所说上屏在主线程的话，我这边的应用整个进程都会展示pag动画，那么这个动画是否会响应应用主线程的任务执行？或者说是否对内存或cpu的负载有影响？感谢~~~

**Reply by kevingpqi123** [reply, marked as answer]
1、对于 PAGView 或 PAGImageView 而言，如果 View 处于不可见的状态，是不会触发渲染的，如果是第一次加载，首帧涉及到构建渲染环境和缓存，会慢一些；
2、上屏只能在主线程哈，子线程无法上屏，你这边如果遇到性能问题，可以把具体的案例发出来，pag 已经发展了这么多年，基础的性能层面我们都考虑到了，你可以在任意的线程中使用 PAGView 或 PAGImageView

**Reply by aSmartTortoise** [reply]
感谢您的回复，上文我描述的概率性PagImageView不可见，不是View的visibility状态，而是动画没加载出来，给人的视觉认识是不可见。另外我再补充一下，我的应用是一个悬浮窗，悬浮窗中的View会加载动画，也就是说在应用整个生命周期这个动画都是running中的，所以担心pag动画是否会有性能方面的问题从而影响主线程任务的执行~

**Reply by kevingpqi123** [reply]
如果可以稳定复现出来，建议基于 pag 的官方 demo 提供复现场景，这样定位是最快的
